File: RotateArray.py
class Solution:
    # @param nums, a list of integer
    # @param k, num of steps
    # @return nothing, please modify the nums list in-place.
     
    def rotate_k(self, nums, k):
        for i in range(k):
            self.rotate_1(nums)
        print nums

    def rotate_1(self, n):
        head = n[0]
        for i in range(len(n) - 1):
            n[i] = n[i+1]
        n[-1] = head

if __name__ == "__main__":
    s = Solution()
    s.rotate_k([1,2,3,4,5,6,7], 2)
    s.rotate_k([1,2,3,4,5,6,7], 4)

File: RotateArray.java
class RotateArray {
	public void rotate_k(int[] nums, int k) {
		for(int i = 0; i < k; i++) {
			rotate_1(nums);
		}
		for(int num : nums)
			System.out.print(num + " ");
		System.out.println();
	}

	public void rotate_1(int[] nums) {
		int head = nums[0];
		for(int i = 0; i < nums.length - 1; i++) {
			nums[i] = nums[i+1];
		}
		nums[nums.length - 1] = head;
	}

	public static void main(String[] args) {
		RotateArray s = new RotateArray();
		int[] nums = {0,1,2,3,4,5};
		s.rotate_k(nums, 2);
		s.rotate_k(nums, 4);
	}
}
